force-app/main/default/classes/ESA_cls_ExternalCommitteeTriggerHandler.cls

```diff
@@ -0,0 +1,61 @@
+/**********************************************************************************************
+* @Author:      Vasanthi D 
+* @Date:        11 Sep 2023
+* @Description: The purpose of this class is to create methods which can be used by different trigger event
+* @Revision(s): [Date] - [Change Reference] - [Changed By] - [Description]   
+***********************************************************************************************/
+public with sharing class ESA_cls_ExternalCommitteeTriggerHandler {
+    public static boolean cloneOpenCases_runsOne = TRUE;
+    public static void cloneOpenCases(List<External_Committee__c> newMeetings){
+        cloneOpenCases_runsOne = FALSE;
+        set<id> parentIds  = new set<id>();
+        Map<id,list<Case>> openCaseMap = new  Map<id,list<Case>>();
+        for(External_Committee__c ec : newMeetings){
+            if(ec.isClone()){
+                parentIds.add(ec.getCloneSourceId());
+            }
+        }     
+        List<Case> caseList = new List<Case>([select id,External_Committee__c,Status from Case where External_Committee__c IN:parentIds AND Status  IN ('Received' ,'Under Review') ]);
+        //retrieve open cases and form the Map 
+        for(Case openCase : caseList){
+            if(openCaseMap.get(openCase.External_Committee__c)==null){
+                openCaseMap.put(openCase.External_Committee__c,new List<Case>{openCase});
+            }
+            else{
+                List<Case> eclist = openCaseMap.get(openCase.External_Committee__c) ;
+                eclist.add(openCase);
+                openCaseMap.put(openCase.External_Committee__c,eclist);   
+            }
+        }     
+        List<Case> caselistToUpdate  = new List<Case>();
+        for(External_Committee__c ec : newMeetings){
```

Comment:
@deepakmulamalla two objects in nested for loop. Why can't you use collections to iterate? This `triggerhandler` will execute both on data loads and manual user creation use-cases and need to be handled appropriately. Also the if statement in line 32 shows `if`, but line 34 shows `If`. Can we check the code before it is sent for review?

Reply:
@cgiSunilM  We have updated the code to resolve this.